src/librustc_infer/infer/combine.rs

```diff
@@ -635,11 +660,21 @@ impl TypeRelation<'tcx> for Generalizer<'_, 'tcx> {
                     }
                 }
             }
+            ty::ConstKind::Unevaluated(..) if self.tcx().features().lazy_normalization_consts => {
+                Ok(c)
+            }
             _ => relate::super_relate_consts(self, c, c),
         }
     }
 }
 
+pub trait ConstEquateRelation<'tcx>: TypeRelation<'tcx> {
```

Comment:
It's not obvious to me at this moment why we have a distinct trait for this

Reply:
There are some`TypeRelation`s which don't need `fn const_equate_obligation` atm, so I ended up adding a bunch of `unimplemented`s while moving `fn const_equate_obligation` to `TypeRelation`. 

I slightly prefer keeping the `ConstEquateRelation` trait for now. Depending on your preference I would drop the last commit of #71973, which merges these two traits. See https://github.com/rust-lang/rust/pull/71973/commits/deaa50e4d322ec547f869dc5cb7efc12a96edce4